oneandone/src/main/java/org/apache/jclouds/oneandone/rest/features/SharedStorageApi.java

```diff
@@ -0,0 +1,114 @@
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import org.apache.jclouds.oneandone.rest.domain.SharedStorage;
+import org.apache.jclouds.oneandone.rest.domain.SharedStorage.Server;
+import org.apache.jclouds.oneandone.rest.domain.SharedStorageAccess;
+import org.apache.jclouds.oneandone.rest.domain.options.GenericQueryOptions;
+import org.apache.jclouds.oneandone.rest.filters.AuthenticateRequest;
+import org.jclouds.Fallbacks;
+import org.jclouds.rest.annotations.BinderParam;
+import org.jclouds.rest.annotations.Fallback;
+import org.jclouds.rest.annotations.MapBinder;
+import org.jclouds.rest.annotations.RequestFilters;
+import org.jclouds.rest.binders.BindToJsonPayload;
+
+@Path("/shared_storages")
+@Produces("application/json")
+@Consumes("application/json")
+@RequestFilters(AuthenticateRequest.class)
+public interface SharedStorageApi extends Closeable {
+
+   @Named("sharedstorages:list")
+   @GET
+   @Fallback(Fallbacks.EmptyListOnNotFoundOr404.class)
+   List<SharedStorage> list();
+
+   @Named("sharedstorages:list")
+   @GET
+   @Fallback(Fallbacks.EmptyListOnNotFoundOr404.class)
+   List<SharedStorage> list(GenericQueryOptions options);
+
+   @Named("sharedstorages:get")
+   @GET
+   @Path("/{sharedStorageId}")
+   @Fallback(Fallbacks.NullOnNotFoundOr404.class)
+   SharedStorage get(@PathParam("sharedStorageId") String sharedStorageId);
+
+   @Named("sharedstorages:create")
+   @POST
+   SharedStorage create(@BinderParam(BindToJsonPayload.class) SharedStorage.CreateSharedStorage sharedStorage);
+
+   @Named("sharedstorages:update")
+   @PUT
+   @Path("/{sharedStorageId}")
+   SharedStorage update(@PathParam("sharedStorageId") String sharedStorageId, @BinderParam(BindToJsonPayload.class) SharedStorage.UpdateSharedStorage sharedStorage);
+
+   @Named("sharedstorages:delete")
+   @DELETE
+   @Path("/{sharedStorageId}")
+   @MapBinder(BindToJsonPayload.class)
+   @Fallback(Fallbacks.NullOnNotFoundOr404.class)
+   SharedStorage delete(@PathParam("sharedStorageId") String sharedStorageId);
+
+   @Named("sharedstorages:servers:list")
+   @GET
+   @Path("/{sharedStorageId}/servers")
+   @Fallback(Fallbacks.EmptyListOnNotFoundOr404.class)
+   List<Server> getServers(@PathParam("sharedStorageId") String sharedStorageId);
+
+   @Named("sharedstorages:servers:create")
+   @POST
+   @Path("/{sharedStorageId}/servers")
+   SharedStorage attachServer(@PathParam("sharedStorageId") String sharedStorageId, @BinderParam(BindToJsonPayload.class) Server.CreateServer server);
+
+   @Named("sharedstorages:servers:get")
+   @GET
+   @Path("/{sharedStorageId}/servers/{serverId}")
+   @Fallback(Fallbacks.NullOnNotFoundOr404.class)
+   Server getServer(@PathParam("sharedStorageId") String sharedStorageId, @PathParam("serverId") String serverId);
+
+   @Named("sharedstorages:servers:delete")
+   @DELETE
+   @Path("/{sharedStorageId}/servers/{serverId}")
+   @MapBinder(BindToJsonPayload.class)
+   @Fallback(Fallbacks.NullOnNotFoundOr404.class)
```

Comment:
Hmmmm I'm not sure about the convenience of this fallback here, as it could be ambiguous. We're not actually deleting something that may not exist, but detaching a resource from one that *should* exist, otherwise the detach operation makes no sense. I'd remove the fallback to let the method propagate an exception on a 404. IMO that would be the less ambiguous feedback for users upon failure. WDYT?